### PR TITLE
[risk=no] [RW-6836] Converting ehr-view component

### PR DESCRIPTION
**Pull request by SrushtiGangireddy**
Converting ehr-view component to react

### CODE REVIEW

File: public-ui/src/app/data-browser/views/ehr-view/ehr-view-react.component.tsx [hunk 1]
```diff
@@ -0,0 +1,650 @@
...
+        this.getDomainTotals();
+   }
+
+   componentDidUpdate(prevProps: Readonly<Props>) {
+        if (prevProps.domainId !== this.props.domainId) {
+            this.getDomainTotals();
+        }
+   }
+
+   changeUrl() {
```
Comment (SrushtiGangireddy) [inline public-ui/src/app/data-browser/views/ehr-view/ehr-view-react.component.tsx:316]: Will change this to route params when converting route.

File: public-ui/src/app/data-browser/views/ehr-view/ehr-view-react.component.tsx [hunk 2]
```diff
@@ -0,0 +1,650 @@
...
+            totalResults: null,
+            concepts: [],
+            standardConcepts: [],
+            currentPage: 1,
+            loading: true,
+            medlinePlusLink: null,
+            medlineTerm: null,
+            selectedMeasurementTypeFilter: false,
+            measurementTestFilter: true,
+            measurementOrderFilter: true,
```
Comment (dolbeew) [inline public-ui/src/app/data-browser/views/ehr-view/ehr-view-react.component.tsx:295]: `measurementTestFilter` and `measurementOrderFilter` are read from state several times but the values never change, they're always `true`. Can we remove them from state or will they change in the future?

File: public-ui/src/app/data-browser/views/ehr-view/ehr-view-react.component.tsx [hunk 3]
```diff
@@ -0,0 +1,650 @@
...
+            showTopConcepts: true,
+            fromDifferentPage: false
```
Comment (dolbeew) [inline public-ui/src/app/data-browser/views/ehr-view/ehr-view-react.component.tsx]: Opposite comment for `fromDifferentPage`. The value is changed but it's never read from state.

File: public-ui/src/app/data-browser/views/ehr-view/ehr-view-react.component.tsx [hunk 4]
```diff
@@ -0,0 +1,650 @@
...
+                            this.setState({
+                                domain: domain,
+                                title: title,
+                                subTitle: subTitle,
+                                totalParticipants: totalParticipants,
+                                numPages: numPages,
+                                totalResults: ehrDomain.standardConceptCount,
+                            }, () => {
+                                this.getTopConcepts();
+                            });
```
Comment (dolbeew) [inline public-ui/src/app/data-browser/views/ehr-view/ehr-view-react.component.tsx]: Nit: these vars only get used once, can be simplified if we drop them:
```suggestion
                            this.setState({
                                domain: domain,
                                title: domain.name,
                                subTitle: 'Keyword: ' + this.props.searchTerm,
                                totalParticipants: domain.participantCount,
                                numPages: Math.ceil(domain.standardConceptCount / 50),
                                totalResults: domain.standardConceptCount,
                            }, () => {
                                this.getTopConcepts();
                            });
```

File: public-ui/src/app/data-browser/views/ehr-view/ehr-view-react.component.tsx [hunk 5]
```diff
@@ -0,0 +1,650 @@
...
+                }).catch(e => {
+                        console.log(e, 'error');
+           });
+   }
+
+   processSearchResults(results) {
+        const {searchWord, top10Results, currentPage, domain} = this.state;
+        results.items.filter(
+            x => PM_CONCEPTS.indexOf(x.conceptId) === -1);
+        results.items.sort((a, b) => b.countValue - a.countValue);
```
Comment (dolbeew) [inline public-ui/src/app/data-browser/views/ehr-view/ehr-view-react.component.tsx]: An array filter will not mutate the original array, so we'll need to reassign it.
Nit: can also combine with the sort by chaining together:
```suggestion
        results.items = results.items
          .filter(x => PM_CONCEPTS.indexOf(x.conceptId) === -1)
          .sort((a, b) => b.countValue - a.countValue);
```

File: public-ui/src/app/data-browser/views/ehr-view/ehr-view-react.component.tsx [hunk 6]
```diff
@@ -0,0 +1,650 @@
...
+        let standardConcepts = [];
+        if (results.standardConcepts) {
+            standardConcepts = results.standardConcepts;
+        }
+        let topResults = top10Results;
+        if (currentPage === 1) {
+            topResults = results.items.slice(0, 10);
+        }
+        this.setState({concepts: results.items, standardConcepts: standardConcepts, top10Results: topResults,
+        loading: false, medlineTerm: medlineTerm, medlinePlusLink: medlinePlusLink});
```
Comment (dolbeew) [inline public-ui/src/app/data-browser/views/ehr-view/ehr-view-react.component.tsx]: Nit: can simplify this by dropping the if statements and setting `standardConcepts` and  `top10Results` inline:
```suggestion
        this.setState({
          concepts: results.items,
          standardConcepts: results.standardConcepts || [],
          top10Results: currentPage === 1 ? results.items.slice(0, 10) : top10Results,
          loading: false,
          medlineTerm: medlineTerm,
          medlinePlusLink: medlinePlusLink
        });
```

File: public-ui/src/app/data-browser/views/ehr-view/ehr-view-react.component.tsx [hunk 7]
```diff
@@ -0,0 +1,650 @@
...
+        } else {
+                this.setState({selectedConcept: concept});
+        }
+    }
+
+    getTopResultsSize() {
+        const {top10Results, title} = this.state;
+        const top10ResultsTitle = (top10Results.length === 1) ? top10Results.length + ' ' + title.slice(0, -1) :
+        (top10Results.length < 10 ? top10Results.length + ' ' + title : 10 + ' ' + title);
+        return top10ResultsTitle;
```
Comment (dolbeew) [inline public-ui/src/app/data-browser/views/ehr-view/ehr-view-react.component.tsx]: Nit: `top10ResultsTitle` var is redundant here, can just return the expression:
```suggestion
        return (top10Results.length === 1) ? top10Results.length + ' ' + title.slice(0, -1) :
        (top10Results.length < 10 ? top10Results.length + ' ' + title : 10 + ' ' + title);
```

File: public-ui/src/app/data-browser/views/ehr-view/ehr-view-react.component.tsx [hunk 8]
```diff
@@ -0,0 +1,650 @@
...
+            minCount: 1,
+            pageNumber: data.selected,
+            measurementTests: measurementTestFilter ? 1 : 0,
+            measurementOrders: measurementOrderFilter ? 1 : 0
+        };
+        let showTopConceptsFlag = true;
+        if (data.selected > 0) {
+            showTopConceptsFlag = false;
+        }
+        this.setState({currentPage: data.selected + 1, showTopConcepts: showTopConceptsFlag});
```
Comment (dolbeew) [inline public-ui/src/app/data-browser/views/ehr-view/ehr-view-react.component.tsx]: Nit: can simplify to a single expression and set `showTopConcepts` inline:
```suggestion
        this.setState({currentPage: data.selected + 1, showTopConcepts: data.selected <= 0});
```

File: public-ui/src/app/data-browser/views/ehr-view/ehr-view-react.component.tsx [hunk 9]
```diff
@@ -0,0 +1,650 @@
...
+                              this.setState({selectedMeasurementTypeFilter: !selectedMeasurementTypeFilter})} />
+                            </button>
+                            <div className='dropdown-menu'>
+                                <div className='clr-checkbox-wrapper'>
+                                    <div className='checkbox-input'>
+                                        <input type='checkbox' id='checkbox1' className='clr-checkbox' onClick={() =>
+                                        this.setState({measurementTestFilter: !measurementTestFilter},
+                                        () => {this.getDomainTotals(); this.getTopConcepts(); } )} defaultChecked={measurementTestFilter}/>
+                                        <label htmlFor='checkbox1' className='checkbox-label'><i className='fas fa-vial fa-rotate-45'
+                                        style={{'transform': 'rotate(315deg)'}}></i> Tests</label>
```
Comment (dolbeew) [inline public-ui/src/app/data-browser/views/ehr-view/ehr-view-react.component.tsx]: Nit: use self-closing syntax for empty `<i>` tag

File: public-ui/src/app/data-browser/views/ehr-view/ehr-view-react.component.tsx [hunk 10]
```diff
@@ -0,0 +1,650 @@
...
+                                    </div>
+                               </div>
+                               <div className='clr-checkbox-wrapper'>
+                                    <div className='checkbox-input'>
+                                        <input type='checkbox' id='checkbox2' className='clr-checkbox' onClick={() =>
+                                        this.setState({measurementOrderFilter: !measurementOrderFilter},
+                                        () => {this.getDomainTotals(); this.getTopConcepts(); } )} defaultChecked={measurementOrderFilter}/>
+                                        <label htmlFor='checkbox2' className='checkbox-label'>
+                                        <i className='far fa-file-signature'></i> Orders</label>
```
Comment (dolbeew) [inline public-ui/src/app/data-browser/views/ehr-view/ehr-view-react.component.tsx]: Nit: use self-closing syntax for empty `<i>` tag

File: public-ui/src/app/data-browser/views/ehr-view/ehr-view-react.component.tsx [hunk 11]
```diff
@@ -0,0 +1,636 @@
...
+                          <div className={dropdownClass}>
+                            <button className='dropdown-toggle'>
+                              <ClrIcon shape='filter-grid' className={filterIconClass} onClick={() =>
+                              this.setState({selectedMeasurementTypeFilter: !selectedMeasurementTypeFilter})} />
+                            </button>
+                            <div className='dropdown-menu'>
+                                <div className='clr-checkbox-wrapper'>
+                                    <div className='checkbox-input'>
+                                        <input type='checkbox' id='checkbox1' className='clr-checkbox' onClick={() =>
+                                        this.setState({measurementTestFilter: !measurementTestFilter},
```
Comment (SrushtiGangireddy) [inline public-ui/src/app/data-browser/views/ehr-view/ehr-view-react.component.tsx:559]: @dolbeew This is the place where the measurementtestfillter/ ordefilter value changes.

File: public-ui/src/app/data-browser/views/ehr-view/ehr-view-react.component.tsx (hunk 3 above)
Comment (SrushtiGangireddy) [reply public-ui/src/app/data-browser/views/ehr-view/ehr-view-react.component.tsx]: Removed this. Not used anymore. Thanks.

File: public-ui/src/app/data-browser/views/ehr-view/ehr-view-react.component.tsx (hunk 4 above)
Comment (SrushtiGangireddy) [reply public-ui/src/app/data-browser/views/ehr-view/ehr-view-react.component.tsx]: Changed this thanks

File: public-ui/src/app/data-browser/views/ehr-view/ehr-view-react.component.tsx (hunk 5 above)
Comment (SrushtiGangireddy) [reply public-ui/src/app/data-browser/views/ehr-view/ehr-view-react.component.tsx]: changed this thanks.

File: public-ui/src/app/data-browser/views/ehr-view/ehr-view-react.component.tsx (hunk 6 above)
Comment (SrushtiGangireddy) [reply public-ui/src/app/data-browser/views/ehr-view/ehr-view-react.component.tsx]: changed this thanks.

File: public-ui/src/app/data-browser/views/ehr-view/ehr-view-react.component.tsx (hunk 7 above)
Comment (SrushtiGangireddy) [reply public-ui/src/app/data-browser/views/ehr-view/ehr-view-react.component.tsx]: changed this thanks.

File: public-ui/src/app/data-browser/views/ehr-view/ehr-view-react.component.tsx (hunk 8 above)
Comment (SrushtiGangireddy) [reply public-ui/src/app/data-browser/views/ehr-view/ehr-view-react.component.tsx]: changed this thanks.

File: public-ui/src/app/data-browser/views/ehr-view/ehr-view-react.component.tsx (hunk 10 above)
Comment (SrushtiGangireddy) [reply public-ui/src/app/data-browser/views/ehr-view/ehr-view-react.component.tsx]: changed this thanks.

**Review by dolbeew — APPROVED** [review]
Sorry, meant to approve this before taking off last week.

**Pull request merged**